elastic/kibana
# [Uptime] Update monitor saved object mappings

## Summary

Fixes https://github.com/elastic/kibana/issues/129424

It updates the default field `name`, `urls`, and `type` mapping from `keyword` to `text`, while adding `.keyword` fields to support sorting by each field.

This also makes importableAndExportable false, since these saved objects contains secrets and import/export without decryption doesn't makes sense. 

### Testing
1. Add the following keys to Kibana.yml https://p.elstc.co/paste/-uRAA0Lw#b4xgrKoWb6yEPQgW+JitoB4WN5eTLV8hLEEaavaV4qA
2. Navigate to Monitor Management. Click `Enable Monitor Management`
3. Add at least 4 monitors, one of each type, with mock data
4. Navigate back to Monitor Management. Ensure that no Kibana errors are present
5. On the Monitor Management List page, click the `Type` column heading. Ensure that monitors are sorted by type
6. Click the `URL` column heading. Ensure the monitors are sorted by `url`. Note: There is a known bug where sorting does not work correctly for tcp and icmp monitors. Issue is pending.
7. Click the `Name` column heading. Ensure the monitors are sorted by name.
8. Navigate to Saved Object Management. Ensure that synthetics monitors are not exportable
9. Search within Saved Objects. Ensure that no Kibana errors are present.
